test/parallel/test-pipe-abstract-socket-http.js

```diff
@@ -0,0 +1,71 @@
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+if (!common.isLinux)
+  common.skip();
+
+const abstractSocket = '\0abstract';
+
+let resStatus;
+
+const server = http.createServer((req, res) => {
+  assert.strictEqual(req.method, 'POST');
+
+  req.resume();
+
+  req.on('end', () => {
+    res.writeHead(200, { 'Content-Type': 'text/plain' });
+    res.end();
+  });
+});
```

Comment:
You might want to use `common` helpers to check whether specific functions are actually called. Here's a suggestion but you get the idea. You can also check other similar tests (maybe the ones starting with `test-http-` )
```suggestion
const server = http.createServer(common.mustCall((req, res) => {
  assert.strictEqual(req.method, 'POST');

  req.resume();

  req.on('end', common.mustCall(() => {
    res.writeHead(200, { 'Content-Type': 'text/plain' });
    res.end();
  }));
}));
```

Reply:
A good resource of how to write a test: https://github.com/nodejs/node/blob/main/doc/contributing/writing-tests.md